### PR TITLE
submit block without vault

### PR DESCRIPTION
**Pull request by InoMurko**
Adding support for block submission if Vault is not present.

### CODE REVIEW

File: apps/engine/lib/engine/ethereum/authority/submitter/external.ex [hunk 1]
```diff
@@ -45,8 +45,19 @@ defmodule Engine.Ethereum.Authority.Submitter.External do
     contract = plasma_framework
 
     fn block_root, nonce, gas_price ->
-      apply(module, function, [block_root, nonce, gas_price, contract, external_opts])
-      :ok
+      case apply(module, function, [block_root, nonce, gas_price, contract, external_opts]) do
+        {:ok, %HTTPoison.Response{body: body, status_code: 200}} ->
+          # ENTERPRISE 1
```
Comment (achiurizo) [inline apps/engine/lib/engine/ethereum/authority/submitter/external.ex:50]: ??? What does this meannn

File: apps/engine/test/engine/db/block_test.exs [hunk 2]
```diff
@@ -698,4 +698,34 @@ defmodule Engine.DB.BlockTest do
         :no_gas_reference
     end
   end
+
+  defp vault_response_mock() do
```
Comment (achiurizo) [inline apps/engine/test/engine/db/block_test.exs:702]: exvcr too heavy?

**Review by achiurizo — APPROVED** [review]

File: .circleci/config.yml [hunk 3]
```diff
@@ -236,9 +236,18 @@ jobs:
           environment:
             ENTERPRISE: 1
           command: mix test.integration_block_submission_vault
-      # - run: rm -rf deps/ _build/
-      # - run: ENTERPRISE=0 mix deps.get
-      # - run: mix test.integration_block_submission
+      - run: rm -rf deps/ _build/
```
Comment (ayrat555) [inline .circleci/config.yml:239]: maybe to run `ENTERPRISE: 0` as a separate step?

File: apps/engine/lib/engine/ethereum/authority/submitter/external.ex (hunk 1 above)
Comment (ayrat555) [reply apps/engine/lib/engine/ethereum/authority/submitter/external.ex:50]: Can `ENTERPRISE 0` and `ENTERPRISE 1` have the same interface? What do you think about wrapping httpoison response and return only tx_hash?

File: apps/engine/test/engine/db/block_test.exs [hunk 4]
```diff
@@ -170,7 +170,7 @@ defmodule Engine.DB.BlockTest do
       # this would be our vault
       integration_point = fn hash, nonce, gas ->
         Kernel.send(parent, {hash, nonce, gas})
-        :ok
+        {:ok, "yolo!"}
```
Comment (ayrat555) [inline apps/engine/test/engine/db/block_test.exs:173]: what do you think about returning data as close to the real one (hash)?

File: apps/engine/test/engine/db/block_test.exs (hunk 2 above)
Comment (ayrat555) [reply apps/engine/test/engine/db/block_test.exs:702]: or you can use FakeServer. It's already added to the childchain

**Review by ayrat555 — APPROVED** [review]

File: .circleci/config.yml (hunk 3 above)
Comment (InoMurko) [reply .circleci/config.yml:239]: step? do you mean job?

File: apps/engine/lib/engine/ethereum/authority/submitter/external.ex (hunk 1 above)
Comment (InoMurko) [reply apps/engine/lib/engine/ethereum/authority/submitter/external.ex:50]: no, they can't have the same interface, if by that, you mean the response?
ENTERPRISE 1 case goes through vault
https://github.com/omgnetwork/submit_block_vault/blob/main/lib/submit_block.ex#L36

ENTERPRISE 0 uses eth send raw transaction https://github.com/omgnetwork/submit_block/blob/master/lib/submit_block.ex#L66

File: apps/engine/test/engine/db/block_test.exs (hunk 2 above)
Comment (InoMurko) [reply apps/engine/test/engine/db/block_test.exs:702]: I don't see a particular reason why I would add another layer of mocking (exvcr or FakeServer).

File: .circleci/config.yml (hunk 3 above)
Comment (ayrat555) [reply .circleci/config.yml:241]: yes

**Pull request merged**